packages/mds-metrics-sheet/metrics-log-utils.ts

```diff
@@ -0,0 +1,191 @@
+import GoogleSpreadsheet from 'google-spreadsheet'
+
+import { promisify } from 'util'
+import log from '@mds-core/mds-logger'
+import { VEHICLE_EVENT, EVENT_STATUS_MAP, VEHICLE_STATUS } from '@mds-core/mds-types'
+import {
+  LastDayStatsResponse,
+  MetricsSheetRow,
+  VehicleCountRow,
+  GoogleSheetCreds,
+  GoogleSheet,
+  GoogleSheetInfo,
+  SpreadsheetWorksheet,
+  ProviderMetrics
+} from './types'
+import { reportProviders, getAuthToken, getVehicleCounts, getLastDayStats } from './shared-utils'
+
+export function eventCountsToStatusCounts(events: { [s in VEHICLE_EVENT]: number }) {
+  return (Object.keys(events) as VEHICLE_EVENT[]).reduce(
+    (acc: { [s in VEHICLE_STATUS]: number }, event) => {
+      const status = EVENT_STATUS_MAP[event]
+      return Object.assign(acc, {
+        [status]: acc[status] + events[event]
+      })
+    },
+    {
+      available: 0,
+      unavailable: 0,
+      reserved: 0,
+      trip: 0,
+      removed: 0,
+      inactive: 0,
+      elsewhere: 0
+    }
+  )
+}
+
+export function sum(arr: number[]) {
+  return arr.reduce((total, amount) => total + (amount || 0))
+}
+
+// Round percent to two decimals
+export function percent(a: number, total: number) {
```

Comment:
Ok, since this is a refactor, this name feels a bit misleading. I would have expected it to be `a/total`. Could this be renamed to something like `complementary_percent`?

Reply:
I think `percent` is a good name here, I'm not sure what `complementary_percent` would indicate differently.

Perhaps something like `convert_fraction_to_percent`, but I'm not crazy about that name either b/c it's not technically a "fraction".

Reply:
Agreed, that's definitely non-intuitive.

Reply:
Done